dropwizard-jetty/src/main/java/io/dropwizard/jetty/ZipExceptionHandlingGzipHandler.java

```diff
@@ -22,12 +24,15 @@
     public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
         try {
             super.handle(target, baseRequest, request, response);
-        } catch (Exception ex) {
-            Throwable rootCause = ExceptionUtils.getRootCause(ex);
-            if (rootCause instanceof ZipException || rootCause instanceof EOFException) {
-                throw new BadMessageException(HttpStatus.BAD_REQUEST_400, rootCause.getMessage(), rootCause);
+        } catch (IOException | ServletException ex) {
+            Optional<BadMessageException> badMessageException = findThrowableInChain(t -> t instanceof ZipException || t instanceof EOFException, ex)
```

Comment:
This is technically a change in behaviour as it finds the first matching exception from the tip of the chain whereas `ExceptionUtils#getRootCause(Throwable)` returns the deepest exception in the chain.

I think in this case the behaviour is now arguably more correct? Is it possible for a ZipException or an EOFException to be in the chain without us wanting to return a 400?